sys/tools/syscalls/core/syscall.lua

```diff
@@ -0,0 +1,444 @@
+                return true
+            -- This system call is some other one line entry.
+            else
+                return true
+            end
+        end
+        return false -- Otherwise, definition added; keep going.
+    end
+    if self:addFunc(line) then
+        return false -- Function added; keep going.
+    end
+    if self:addArgs(line) then
+        return false -- Arguments added; keep going.
+    end
+    return self:isAdded(line) -- Final validation, before adding.
+end
+
+-- Return TRUE if this system call is native.
+-- NOTE: The other system call names are also treated as native, so that's why
+-- they're being allowed in here.
+function syscall:native()
+    return self:compat_level() == native or self.name == "lkmnosys" or
+           self.name == "sysarch"
+end
+
+function syscall:new(obj)
+	obj = obj or { }
+	setmetatable(obj, self)
+	self.__index = self
+
+	self.expect_rbrace = false
+    self.changes_abi = false
+	self.args = {}
+
+	return obj
+end
+
+-- Make a shallow copy of `self` and replace the system call number with num
+-- (which should be a number).
+-- For system call ranges.
+function syscall:shallowCopy(num)
+	local obj = syscall:new()
+
+	-- shallow copy
+	for k, v in pairs(self) do
+		obj[k] = v
+	end
+	obj.num = num	-- except override range
+	return obj
+end
+
+-- Make a deep copy of the parameter object.
+-- CREDIT: http://lua-users.org/wiki/CopyTable
+-- For a full system call (the nested arguments table should be a deep copy).
+local function deepCopy(orig)
+    local type = type(orig)
+    local copy
+
+    if type == 'table' then
+        copy = {}
+        for orig_key, orig_value in next, orig, nil do
+            copy[deepCopy(orig_key)] = deepCopy(orig_value)
+        end
+        setmetatable(copy, deepCopy(getmetatable(orig)))
+    else -- number, string, boolean, etc
+        copy = orig
+    end
+
+    return copy
+end
+
+-- CREDIT: http://lua-users.org/wiki/CopyTable
+-- Save copied tables in `copies`, indexed by original table.
+function deepCopy(orig, copies)
+    copies = copies or {}
+    local orig_type = type(orig)
+    local copy
+    if orig_type == 'table' then
+        if copies[orig] then
+            copy = copies[orig]
+        else
+            copy = {}
+            copies[orig] = copy
+            for orig_key, orig_value in next, orig, nil do
+                copy[deepCopy(orig_key, copies)] = deepCopy(orig_value, copies)
+            end
+            setmetatable(copy, deepCopy(getmetatable(orig), copies))
+        end
+    else -- number, string, boolean, etc
+        copy = orig
+    end
+    return copy
+end
+
+--
+-- As we're parsing the system calls, there's two types. Either we have a
+-- specific one, that's a assigned a number, or we have a range for things like
+-- reseved system calls. this function deals with both knowing that the specific
+-- ones are more copy and so we should just return the object we just made w/o
+-- an extra clone.
```

Comment:
I'm not sure what this sentence is trying to say.

Reply:
I've reworded. Please let me know if it is still unclear.